caldav/objects.py

```diff
@@ -47,6 +48,50 @@ def errmsg(r):
     return "%s %s\n\n%s" % (r.status, r.reason, r.raw)
 
 
+def _expand_event(event, start, end):
```

Comment:
Perhaps it could be a class method?  `def expand_rrule(self, start, end)`, perhaps under the `CalendarResourceObject` class?  (that's the parent class for Todo, Event, Journal)